### PR TITLE
[NUI] Fix MatchParent to fill remaining space in LinearLayout

### PR DESCRIPTION
**Pull request by jaehyun0cho**
Previously, in LinearLayout, LayoutParamPolicies.MatchParent sets child
size to its parent size.

Now, in LinearLayout, LayoutParamPolicies.MatchParent calculates the
remaining space of parent and fills child in the remaining space of
parent.

There are 2 cases to fill child in the remaining space of parent.
1. MatchParent (both with or without weight)
 - If child has MatchParent specification, then child is filled in the
   remaining space of parent.
2. Specification 0 with weight
 - If child has zero specification and weight is greater than 0, then
   child is filled in the remaining space of parent proportional to its
   weight.

### API Changes ###

 - ACR:

### CODE REVIEW

File: src/Tizen.NUI/src/public/Layouting/LinearLayout.cs [hunk 1]
```diff
@@ -268,88 +243,68 @@ private void MeasureHorizontal(MeasureSpecification widthMeasureSpec, MeasureSpe
             var widthMode = widthMeasureSpec.Mode;
             var heightMode = heightMeasureSpec.Mode;
             bool isExactly = (widthMode == MeasureSpecification.ModeType.Exactly);
-            bool matchHeight = false;
+            bool isHeightExactly = (heightMode == MeasureSpecification.ModeType.Exactly);
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/Layouting/LinearLayout.cs:246]: could you check the VS warning below?

_**src/Tizen.NUI/src/public/Layouting/LinearLayout.cs#L247
The variable 'allFillParent' is assigned but its value is never used**_

Comment (jaehyun0cho) [reply src/Tizen.NUI/src/public/Layouting/LinearLayout.cs:246]: Thank you for your comment.
The unused variable has been removed.

**Review by dongsug-song — COMMENTED** [review]
looks good to me, but other member's review is required.

**Review by everLEEst — COMMENTED** [review]
Looks good to me except one old use case.

in old use case,
add content with MatchParent param in ScrollableBase
will create N-th Page Scrollable View.

can be guide them proper solution after this patch merged.

**Comment by jaehyun0cho** [comment]
> Looks good to me except one old use case.
> 
> in old use case,
> add content with MatchParent param in ScrollableBase
> will create N-th Page Scrollable View.
> 
> can be guide them proper solution after this patch merged.

As @everLEEst  and I discussed later, this case is not a problem case.
The reason as follows.

1. User defined ScrollableBase case
    - Suppose that user defined ScrollableBase uses LinearLayout.
      i. If ScrollableBase's ContentContainer uses WrapContent, then children cannot use MatchParent. (circular dependency)
      ii. If ScrollableBase's ContentContainer uses MatchParent or specific size and children uses MatchParent, then children's size fills ScrollableBase's ContentContainer, so scrolling cannot work.

2. Default ScrollableBase case
    - Default ScrollableBase's ContentContainer does not use Layout. Therefore, children cannot use MatchParent/WrapContent to set their position.

As a result, this PR does cause the above backward compatibility issue.

**Review by dongsug-song — APPROVED** [review]
looks good to me

**Pull request merged**